### PR TITLE
install-nix: Solve #971 USER == NULL

### PR DESCRIPTION
**Pull request by Anton-Latukha**
This addresses #971.

Which also encountered by me:

```bash
docker run -it alpine env
docker run -it alpine /bin/sh -c 'echo "$USER <-this" ; whoami'
docker run -it ubuntu /bin/sh -c 'echo "$USER <-this" ; whoami'
docker run -it base/archlinux /bin/sh -c 'echo "$USER <-this" ; whoami'
docker run -it nixos/nix /bin/sh -c 'echo "$USER <-this" ; whoami'
```

This is because Docker env not goes through `login` and login shell, so shell not sources login config, so shell not defined that env variables.

We touch `USER` only when it is not defined. And as it is shell, anyway it is a copy variable in scope of this script.

### CODE REVIEW

**Comment by grahamc** [comment]
lgtm

**Comment by Anton-Latukha** [comment]
Does in project flow is to squash on merge, - or do I need to do squashing/acceptance squashing.

**Comment by wmertens** [comment]
LGTM. @Anton-Latukha maybe you want to squash it all into a single commit.

**Comment by Anton-Latukha** [comment]
That is what I asked above.

Is there is no squash on merge enabled and no workflow to squash on merge in repository.

It seems life the natural way to move.

It is nature of someone who contribute pull request and waits on acceptance - to better it's pull request. To make fixes, add comments, polish code and text.
With squash on pull request contributor - he needs to squash all commits every time, at every fix, at every improvement, to keep things clean and ready to merge.

With squash on merge on maintainer side:
1. Maintainer doesn't need to write to me that I need to make squash.
2. Maintainer doesn't need to wait when and do I going to do squash.
3. Maintainer can chose by himself the name of commit. So standardized commit names can be enforced.
4. All can be done as maintainer wants, when maintainer wants.
5. It is already a GUI feature on GitHub. https://help.github.com/articles/merging-a-pull-request/#merging-a-pull-request-on-github

File: scripts/install-nix-from-closure.sh [hunk 1]
```diff
@@ -2,19 +2,21 @@
...
 if ! [ -e "$self/.reginfo" ]; then
     echo "$0: incomplete installer (.reginfo is missing)" >&2
 fi
 
+# in case if USER not set
 if [ -z "$USER" ]; then
-    echo "$0: \$USER is not set" >&2
-    exit 1
+    echo "$0: Environment variable USER is not set" >&2
+    USER="$(whoami)"    # Taking username from  UID
```
Comment (domenkozar) [inline scripts/install-nix-from-closure.sh]: Shouldn't that also be readonly?

**Comment by Anton-Latukha** [comment]
I was solving the real problem.

Most of scripts was in very poor state. And they do not have constants used. And they still do not.

I needed to merge at least something to have a progress in master.
To continue the work. To know that my things get accepted.

I offered myself to be the guy who would make a great quality Nix install scrips, and who would maintains it.
My experience in community is great, except this Nix scripts.

If I can not get a tiniest changes merged. Then, what can I do about that.

Now it can be cleaned up.

Anyway. Thanks to everybody.

**Pull request closed without merging**